src/main/java/io/camunda/zeebe/process/test/testengine/RecordStreamSourceImpl.java

```diff
@@ -33,8 +33,8 @@
 
   private final LogStreamReader logStreamReader;
   private final int partitionId;
-  private List<Record<?>> records = new ArrayList<>();
-  private long lastPosition = -1L;
+  private volatile List<Record<?>> records = new ArrayList<>();
```

Comment:
:x:
```suggestion
  private final List<Record<?>> records = new ArrayList<>();
```
This field is never written to.